### ISSUE TITLE
SMS verification request before 'expiresIn' has reached the limit

### ISSUE THREAD

**Issue by fjehlik** [jmfernandes/robin_stocks#287]
Howdy Folks,

After logging into Robinstocks, after a number of days it I get an SMS error and it stops collecting information. To correct this I have to log back in and submit a new SMS verification. I have 'expiresIn' set to a month, 86400s/day * 30 days, as shown below/ I thought this would take care of the SMS request for 30 days. However, it appears this happens now about once a week rather than the month. 

Anyone else had this issue or know a way around this? I'd like the trading bot to continue uninterrupted. Any info is appreciated~

login = rs.login(username=name,
          password=passcode,
          expiresIn=86400*30,  # 86400 = s/day
          by_sms=True,
          store_session=True)

**Comment by fjehlik** [comment]
Hey thanks for the response. I have not done so, I currently only log in at the initiation of the program and let it run. I assumed the expiresIn period would keep it logged in until the expiration time was met.

**Comment by aaroncouch** [comment]
This sounds like a limitation set by Robinhood themselves, but I could be wrong. I can try testing. Just curious when you login and get the returned json what's set in the `expires_in` key? Is it set appropriately to one month?

**Comment by fjehlik** [comment]
That's a great question, I'll be happy to check. By any chance do you know the default location the json file gets stored (or know the name)? Using Linux (Ubuntu), did a quick json search, tons of stuff to wade through.

**Comment by aaroncouch** [comment]
That information is pickled in `~/.tokens/robinhood.pickle` obviously do not paste the output here. However when you execute the rh.login method you should be returned a dictionary of the resulting JSON response from Robinhood's `/oauth2/token/` endpoint. That's what I'm curious in. How does their API respond to your expiration.

**Comment by fjehlik** [comment]
I just logged in a separate instance. In the dictionary that python outputs, the 'expires_in' is 2592000 (which is a month in seconds). Under 'detail' it logged in using authentication in robinhood.pickle. I'll keep my eye on it tomorrow to see if i get the error again (which is a week since the program was initiated). Your feedback has been greatly appreciated. High five~

**Comment by fjehlik** [comment]
It's been a little while so I'll update. It is still logging out after a week rather than the month set in the **expires_in** key. To correct I am forced to log back in and send another SMS authentication. 

@aaroncouch, you mentioned using an 8 hr login method in your bot. Do you have SMS off so that you don't have to enter that every 8 hours? Example below:

login = rs.login(username=config.credential['username'],
          password=config.credential['passcode'],
          expiresIn=86400*30,  # 86400 = s/day
          **by_sms=False,**
          store_session=True) 


In my case, **by_sms=True**

**Comment by aaroncouch** [comment]
That's right. Just like you said, regardless of my actions the token expires every 7 days. I find this very interesting considering when I auth via the webapp or phone app I only auth the device maybe once a month. Perhaps they have rules based on device type (user agent?). But yeah, my bot has a re-auth method that I set at specific times in the day (when I am awake). The `by_sms=False" method will have your auth code sent via email. There could be a potential solution there, but would require engineering work.

**Comment by fjehlik** [comment]
I have the same on my side. Logging in on my phone app I haven't had to verify via sms in a long time and you may well be right, perhaps it is device dependent. Coding a solution to this would be great, unfortunately it would require quite a bit of relatively complex work me thinks. I appreciate all the input~
